File: 프로그래머스/unrated/181927. 마지막 두 원소/마지막 두 원소.java
class Solution {
    public int[] solution(int[] num_list) {
        int[] answer = {};
        
        answer=new int[num_list.length+1];
        
        
        //마지막 원소의 인덱스
        int last=num_list.length-1;
        
        //마지막 원소가 그 전 원소보다 큰 경우
        if(num_list[last]>num_list[last-1]){
            answer[last+1]=num_list[last]-num_list[last-1];
        
        //마지막 원소가 그 전 원소보다 작은 경우   
        }else{
            answer[last+1]=num_list[last]*2;
        }
        
        for(int i=0; i<num_list.length; i++){
            answer[i]=num_list[i];
        }
        
        return answer;
    }
}

File: 프로그래머스/0/181927. 마지막 두 원소/마지막 두 원소.py
def solution(num_list):
    if len(num_list) >= 2:
        last_element = num_list[-1]
        second_last_element = num_list[-2]

        if last_element > second_last_element:
            answer = num_list + [last_element - second_last_element]
        else:
            answer = num_list + [last_element * 2]
    else:
        # 리스트의 길이가 1 이하일 경우, 그대로 반환
        answer = num_list.copy()

    return answer
